### ISSUE TITLE
NiftyNet retraining

### ISSUE THREAD

**Issue by jordandekraker** [khanlab/hippunfold#7]
Generate snakemake workflow to train or fine-tune new NiftyNet models (sometimes also referred to as highres3dnet model, UNet model, CNN model. It is in fact all of these things).

**Comment by akhanf** [comment]
how about if we made a snakemake rule to gather all the niftynet_lbl.nii.gz and img.nii.gz files and package into single folder (with the desired file naming)? then user could edit labels as needed, then invoke another rule to package them up and create a niftynet config file for re-training (or just submit the training job directly)?

**Comment by akhanf** [comment]
btw looks like we could use https://pypi.org/project/ConfigArgParse/ to read, edit and write niftynet config files easily

**Comment by jordandekraker** [comment]
Just added this script to do this: https://github.com/khanlab/hippunfold/blob/master/resources/retrain_UNet.py
I'll still have to set it up to run as a function and make sure dependencies are in place, and I'll add to the documentation explaining how this should be set up.

A few comments I'm still wondering about:
1) would it make sense to set this up as a snakemake rule? I'm thinking probably not since its not expected to be used in all cases
2) dependencies are NiftyNet v0.5 (haven't tested backward compatibility from v0.6) with tensorflow backend, and ConfigArgParse. Not sure if these would be best handled in python or in a singularity container, but I haven't yet been able to get this working on graham outside of a container. 
3) not sure how to call niftynet's `net_segment` from within a python function, and I'd rather not add another wrapper. I'm leaning towards `import subprocess`
4) still haven't written a script to overwrite niftynet's usual dataset_split.csv. This will need to be done to ensure unique subject IDs are not split across train/test/validation (but honestly it doesn't make much difference)

**Comment by akhanf** [comment]
closing this, since this is no longer an issue with the move to nnU-net
